test/insertion_order_parser_test.dart

```diff
@@ -0,0 +1,302 @@
+          '"automationType":"INSERTION_ORDER_AUTOMATION_TYPE_NONE",'
+          '$budgetSegment}';
+      input = '{$advertiserId, $campaignId, $insertionOrderId, $displayName,'
+          '$entityStatus, $updateTime, $pacing, $budget}';
+
+      final actual = InsertionOrderParser.parse(input);
+
+      final expectedPacing = InsertionOrder_Pacing()
+        ..pacingPeriod = InsertionOrder_Pacing_PacingPeriod.PACING_PERIOD_FLIGHT
+        ..pacingType = InsertionOrder_Pacing_PacingType.PACING_TYPE_AHEAD
+        ..dailyMaxImpressions = emptyEntry;
+      final firstSegment =
+          InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment()
+            ..budgetAmountMicros = '4000000'
+            ..description = 'year-2019'
+            ..campaignBudgetId = emptyEntry
+            ..dateRange = oneDateRange;
+      final secondSegment =
+          InsertionOrder_InsertionOrderBudget_InsertionOrderBudgetSegment()
+            ..budgetAmountMicros = '2000000'
+            ..description = emptyEntry
+            ..campaignBudgetId = emptyEntry
+            ..dateRange = oneDateRange;
+      final expectedBudget = InsertionOrder_InsertionOrderBudget()
+        ..budgetUnit =
+            InsertionOrder_InsertionOrderBudget_BudgetUnit.BUDGET_UNIT_CURRENCY
+        ..automationType =
+            InsertionOrder_InsertionOrderBudget_InsertionOrderAutomationType
+                .INSERTION_ORDER_AUTOMATION_TYPE_NONE
+        ..budgetSegments.addAll([firstSegment, secondSegment]);
+      final expected = InsertionOrder()
+        ..advertiserId = '11111'
+        ..campaignId = '2222222'
+        ..insertionOrderId = '3333333'
+        ..displayName = 'display name'
+        ..updateTime = '2020-06-23T17:14:58.685Z'
+        ..entityStatus = InsertionOrder_EntityStatus.ENTITY_STATUS_ACTIVE
+        ..pacing = expectedPacing
+        ..budget = expectedBudget;
+      expect(actual, expected);
+    });
+
+    group('parse Pacing from json that contains:', () {
+      test('nothing', () {
+        input = '{"pacing":{}}';
+
+        final actual = InsertionOrderParser.parse(input);
+
+        expect(actual, insertionOrderTemplate);
+      });
+
+      test('pacingPeriod and pacingType', () {
+        input = '{"pacing":{"pacingPeriod":"PACING_PERIOD_FLIGHT",'
+            '"pacingType":"PACING_TYPE_AHEAD"}}';
+
+        final actual = InsertionOrderParser.parse(input);
+
+        final pacing = InsertionOrder_Pacing()
+          ..pacingPeriod =
+              InsertionOrder_Pacing_PacingPeriod.PACING_PERIOD_FLIGHT
+          ..pacingType = InsertionOrder_Pacing_PacingType.PACING_TYPE_AHEAD
+          ..dailyMaxImpressions = emptyEntry;
+        final expected = insertionOrderTemplate..pacing = pacing;
+        expect(actual, expected);
+      });
+
+      test('pacingPeriod, pacingType and dailyMaxMicros', () {
+        input = '{"pacing":{"pacingPeriod":"PACING_PERIOD_FLIGHT",'
```

Comment:
Are you sure this is what the service returns? When we make API calls from the DV360 Web UI the protos are all converted to 'lite' json which uses numbers to return the enum values instead of strings.

Reply:
Yes, that's what get returned when I use the google javascript client library to make those API calls. I can double check this with the API team.